app/test/unit/branch-pruner-test.ts

```diff
@@ -0,0 +1,261 @@
+import * as moment from 'moment'
+import { BranchPruner } from '../../src/lib/stores/helpers/branch-pruner'
+import { Repository } from '../../src/models/repository'
+import { GitStoreCache } from '../../src/lib/stores/git-store-cache'
+import { RepositoriesStore, GitStore } from '../../src/lib/stores'
+import { RepositoryStateCache } from '../../src/lib/stores/repository-state-cache'
+import { setupFixtureRepository } from '../helpers/repositories'
+import { shell } from '../helpers/test-app-shell'
+import { TestRepositoriesDatabase } from '../helpers/databases'
+import { GitProcess } from 'dugite'
+import { IGitHubUser } from '../../src/lib/databases'
+import { IAPIRepository } from '../../src/lib/api'
+
+describe('BranchPruner', () => {
+  const onGitStoreUpdated = () => {}
+  const onDidLoadNewCommits = () => {}
+  const onDidError = () => {}
+
+  let gitStoreCache: GitStoreCache
+  let repositoriesStore: RepositoriesStore
+  let repositoriesStateCache: RepositoryStateCache
+  let onPruneCompleted: jest.Mock<(repository: Repository) => Promise<void>>
+
+  beforeEach(async () => {
+    gitStoreCache = new GitStoreCache(
+      shell,
+      onGitStoreUpdated,
+      onDidLoadNewCommits,
+      onDidError
+    )
+
+    const repositoriesDb = new TestRepositoriesDatabase()
+    await repositoriesDb.reset()
+    repositoriesStore = new RepositoriesStore(repositoriesDb)
+    repositoriesStateCache = new RepositoryStateCache(
+      () => new Map<string, IGitHubUser>()
+    )
+    onPruneCompleted = jest.fn(() => (_: Repository) => {
+      return Promise.resolve()
+    })
+  })
+
+  it('Does nothing on non GitHub repositories', async () => {
```

Comment:
:art: we usually avoid sentence case for these test names. These were the only exceptions I could find:

<img width="412" alt="screen shot 2019-02-06 at 10 26 21 am" src="https://user-images.githubusercontent.com/359239/52348023-ba421800-29f9-11e9-9049-da017d6d6f18.png">

If you run `yarn test:unit --verbose` you'll see what this looks like:

```
...

 PASS  app/test/unit/list-selection-test.ts
  list-selection
    findNextSelectableRow
      ✓ returns first row when selecting down outside list (filter text) (2ms)
      ✓ returns first selectable row when header is first (1ms)
      ✓ returns first row when selecting down from last row (1ms)
      ✓ returns last row when selecting up from top row

 PASS  app/test/unit/cloning-repository-test.ts
  CloningRepository
    name
      ✓ provides the name of the repository being cloned (1ms)
      ✓ extracts the repo name from the url not the path
      ✓ extracts the repo name without git suffix (1ms)

 PASS  app/test/unit/fatal-error-test.ts
  forceUnwrap
    ✓ fails for null (1ms)
    ✓ fails for undefined (1ms)
    ✓ passes for false
    ✓ passes for a hash (1ms)
```

```suggestion
  it('does nothing on non GitHub repositories', async () => {
```